Source/Core/SwipeActions.swift

```diff
@@ -0,0 +1,168 @@
+//
+//  Swipe.swift
+//  Eureka
+//
+//  Created by Marco Betschart on 14.06.17.
+//  Copyright © 2017 Xmartlabs. All rights reserved.
+//
+
+import Foundation
+
+public typealias SwipeActionHandler = (ContextualAction, ContextualActionSource, ((Bool) -> Void)?) -> Void
+
+public class SwipeAction{
+	public let contextualAction: ContextualAction
+	
+	@available(iOS 11, *)
+	public var backgroundColor: UIColor?{
+		get{
+			guard let contextualAction = self.contextualAction as? UIContextualAction else{ return nil }
+			return contextualAction.backgroundColor
+		}
+		set{
+			guard let contextualAction = self.contextualAction as? UIContextualAction else{ return }
+			contextualAction.backgroundColor = newValue
+		}
+	}
+	
+	@available(iOS 11, *)
+	public var image: UIImage?{
+		get{
+			guard let contextualAction = self.contextualAction as? UIContextualAction else{ return nil }
+			return contextualAction.image
+		}
+		set{
+			guard let contextualAction = self.contextualAction as? UIContextualAction else{ return }
+			return contextualAction.image = newValue
+		}
+	}
+	
+	public var title: String?{
+		get{
+			if #available(iOS 11, *), let contextualAction = self.contextualAction as? UIContextualAction{
+				return contextualAction.title
+			
+			} else if let contextualAction = self.contextualAction as? UITableViewRowAction{
+				return contextualAction.title
+			}
+			return nil
+		}
+		set{
+			if #available(iOS 11, *), let contextualAction = self.contextualAction as? UIContextualAction{
+				contextualAction.title = newValue
+			
+			} else if let contextualAction = self.contextualAction as? UITableViewRowAction{
+				contextualAction.title = newValue
+			}
+		}
+	}
+	
+	public let handler: SwipeActionHandler
+	public let style: Style
+	
+	public init(style: Style, title: String?, handler: @escaping SwipeActionHandler){
+		self.style = style
+		self.handler = handler
+		
+		if #available(iOS 11, *){
+			self.contextualAction = UIContextualAction(style: style.contextualStyle as! UIContextualAction.Style, title: title){ action, view, completion -> Void in
+				handler(action, view, completion)
+			}
+			
+		} else {
+			self.contextualAction = UITableViewRowAction(style: style.contextualStyle as! UITableViewRowActionStyle,title: title){ (action, indexPath) -> Void in
+				handler(action, indexPath, nil)
+			}
+		}
+	}
+	
+	public enum Style: Int{
```

Comment:
unnecessary `: Int` here.

Reply:
Compiler throws `Enum case cannot have a raw value if the enum does not have a raw type` if `: Int` is removed